app/javascript/src/components/editor/EditorWiki.svelte

```diff
@@ -3,15 +3,31 @@
   import EditorWikiSearch from "./EditorWikiSearch.svelte"
 
   let article
+  let contentElement
 
   function click(event) {
     if (!event.target.href) return
 
     event.preventDefault()
 
+    const href = new URL(event.target.href)
+
     const currentHost = window.location.hostname
-    if (event.target.href.includes(currentHost)) fetchArticle(event.target.href)
-    else window.open(event.target.href, "_blank")
+    if (href.hostname === currentHost) {
+      if (!!href.hash) {
+        jumpToSection(href.hash.substring(1))
+        return
+      } else if (href.pathname.startsWith("/wiki")) {
+        fetchArticle(event.target.href)
+        return
+      }
+    }
+
+    window.open(event.target.href, "_blank")
```

Comment:
If we invert this check we could use an early return and indent less

```suggestion
    if (href.hostname != currentHost) {
      window.open(event.target.href, "_blank")
      return
    }
    
    if (!!href.hash) jumpToSection(href.hash.substring(1))
    else if (href.pathname.startsWith("/wiki")) fetchArticle(event.target.href)
```